pyrolite/mineral/normative.py

```diff
@@ -153,213 +158,250 @@ def CIPW_norm(data, form="weight"):
     This function is currently a stub.
     """
     columns = ['SiO2','TiO2','Al2O3','Fe2O3','FeO','MnO','MgO','CaO','Na2O','K2O','P2O5','CO2','SO3']
-    mol_weights = pd.Series([60.08,79.90,101.96,159.6882,71.844,70.94,40.30,56.08,61.98,94.20,141.94, 44.01, 80.06], index = columns)
+    mol_w = pd.Series([pt.formula(c).mass for c in columns], index=columns)
 
     # If want to convert to FeOtotal
 #     data['FeO'] = data['FeO'] + (data['Fe2O3']*(2*71.8444/159.6882))
 #     data['Fe2O3'] = 0
 
     # Normalise to 100 on anhydrous basis
+    print(data)
+#     res = data[columns].copy(deep = True).pyrochem.to_molecular()
```

Comment:
Having issues with this for some reason, it doesn't see to convert to molecular

Reply:
Had a quick look and there was indeed something strange going on here, potentially due to how assignment was used in `pyrolite.geochem.__init__.py`. Your code achieves the same thing, so keep this for now! I've made a change which seems to fix this on develop, and we can change it later if desired.  There were one or two other functions potentially with the same issue, but these specific dataframe accessor methods aren't used elsewhere in the package so it seems contained to there.